### ISSUE TITLE
Dependencies consistency with JupyterLite

### ISSUE THREAD

**Issue by JohanMabille** [jupyterlite/jupyterlite#926]
When loading a jupyter lite kernel, we need to ensure that the jupyterlite version used to run the kernel (or at least the versions of jupyterlite packages which are dependencies of the lite kernel) is the same as that used to build the lite kernel, otherwise the kernel crashes.

**Comment by jtpio** [comment]
Thanks @JohanMabille for the report.

I'm curious to know in which situation this happened? Normally federated kernels (like `xeus-python`) depend on on the `@jupyterlite/kernel` and `@jupyterlite/server` packages to compile the extension.

Maybe this is related to a newer JupyterLite release which had breaking changes, while the kernel was still consuming interfaces for a previous version? The changelog is available here: https://jupyterlite.readthedocs.io/en/latest/changelog.html

> otherwise the kernel crashes

Crashes like a hard crash? Is there any error messages in the dev tools console?

**Comment by martinRenou** [comment]
I don't have the exact context, so sorry if I'm not saying something very accurate.

My gut feeling is that something may have changed in the `DriveFS` APIs that jupyterlite-xeus-python consumes, and broke it?

**Comment by JohanMabille** [comment]
> Maybe this is related to a newer JupyterLite release which had breaking changes, while the kernel was still consuming interfaces for a previous version? 

Yes exactly, the version of jupyterlite installed with pip was 0.1.0-beta12 while the dependency in `package.json` was `0.1.0-alpha14`.

> Crashes like a hard crash?

I don't remember the detail, but it was not a hard crash of the browser. And the error was not obvious from the message in the console.

**Comment by bollwyvl** [comment]
Yep. We're still having to change apis. Take that b in the version
seriously, pin hard.

**Comment by jtpio** [comment]
> Yes exactly, the version of jupyterlite installed with pip was 0.1.0-beta12 while the dependency in `package.json` was `0.1.0-alpha14`.

For now it's probably better to target the latest release, by pinning both to the latest Python package in the dev environment (for example `jupyterlite==0.1.0b16`), and the JS packages (for example `"@jupyterlite/kernel": "^0.1.0-beta.16"`).

**Comment by JohanMabille** [comment]
So just to be clear, the problem is not that we have to hard pin, or which version to pin, ut rather the silent failure when pinning a version of jupyterlite different from that installed. I juste reproduced it, the result is jupyterlite can never conect to the kernel, and there is no meaningful message in the console:

![jupyterlite](https://user-images.githubusercontent.com/6754742/208958630-2d43d7d7-bd37-42cb-88f1-63ebcd7953e9.png)

**Comment by martinRenou** [comment]
Is there something in the JavaScript console?

**Comment by jtpio** [comment]
On the screenshot above it looks like the kernel is in the "Connecting" state?

Agree that checking the dev tools console would be helpful.

**Comment by bollwyvl** [comment]
Yeah, some browser console log, and some light breakpoint trawling by someone familiar with what it's expecting would be very helpful.

Do the changes in https://github.com/jupyterlite/jupyterlite/pull/900 play into this?

And for something completely different: any feedback on https://github.com/jupyterlite/jupyterlite/issues/929 (addons can specify short CLI args) would be helpful.
